spec/realtime/auth.test.js

```diff
@@ -602,6 +602,17 @@ define(['ably', 'shared_helper', 'async', 'chai'], function (Ably, helper, async
 			)
 		); /* expectFailed: */
 
+		/* 403 should cause connection to fail even with an external error response */
+		it(
+			'authUrl_403_custom_error',
+			authCallback_failures(
+				{
+					authUrl: echoServer + '/respondwith?status=403&type=json&body={"error"={"some_custom": "error"}}'
```

Comment:
> `'/respondwith?status=403&type=json&body={"error"={"some_custom": "error"}}'`

- the `/respondwith` endpoint doesn't return a response body. your echoserver pr went the route of adding status support to `/` instead
- `{"error"={` is malformed, equals instead of colon
- the json in the querystring needs to be uri-encoded. eg `'/?status=403&type=json&body=' + encodeURIComponent(JSON.stringify({error: {some_custom: "error"}}))`

Reply:
Whoops. I've gone with your suggestion in ed8be29.